src/kafkerl_connector.erl

```diff
@@ -295,41 +306,41 @@ request_metadata(Pid, Brokers, TCPOpts, Retries, RetryInterval, Request) ->
                      -1 -> -1;
                      N  -> N - 1
                    end,
-      request_metadata(Pid, Brokers, TCPOpts, NewRetries, RetryInterval,
+      do_request_metadata(Pid, Brokers, TCPOpts, NewRetries, RetryInterval,
                        Request)
   end.
 
-request_metadata([], _TCPOpts, _Request) ->
+do_request_metadata([], _TCPOpts, _Request) ->
   {error, all_down};
-request_metadata([{Host, Port} = _Broker | T] = _Brokers, TCPOpts, Request) ->
+do_request_metadata([{Host, Port} = _Broker | T] = _Brokers, TCPOpts, Request) ->
   lager:debug("Attempting to connect to broker at ~s:~p", [Host, Port]),
   % Connect to the Broker
   case gen_tcp:connect(Host, Port, TCPOpts) of
     {error, Reason} ->
       warn_metadata_request(Host, Port, Reason),
       % Failed, try with the next one in the list
-      request_metadata(T, TCPOpts, Request);
+      do_request_metadata(T, TCPOpts, Request);
     {ok, Socket} ->
       % On success, send the metadata request
       case gen_tcp:send(Socket, Request) of
         {error, Reason} ->
           warn_metadata_request(Host, Port, Reason),
           % Unable to send request, try the next broker
-          request_metadata(T, TCPOpts, Request);
+          do_request_metadata(T, TCPOpts, Request);
         ok ->
           case gen_tcp:recv(Socket, 0, 6000) of
```

Comment:
The expression on line 331 and column 11 is nested beyond the maximum level of 3.